src/Compilers/CSharp/Portable/Symbols/Tuples/TupleTypeSymbol.cs

```diff
@@ -109,7 +109,7 @@ internal sealed class TupleTypeSymbol : WrappedNamedTypeSymbol
             }
 
             var constructedType = Create(underlyingType, elementNames, errorPositions, locationOpt, elementLocations);
-            if (shouldCheckConstraints)
+            if (shouldCheckConstraints && diagnostics != null)
```

Comment:
> && diagnostics != null [](start = 38, length = 23)

This doesn't feel right, we might be suppressing errors that have nothing to do with availability of the type.